docs/reference/analysis/charfilters/pattern-replace-charfilter.asciidoc

```diff
@@ -1,37 +1,296 @@
+          "char_filter": [
+            "my_char_filter"
+          ],
+          "filter": [
+            "lowercase"
+          ]
+        }
+      },
+      "char_filter": {
+        "my_char_filter": {
+          "type": "pattern_replace",
+          "pattern": "(?<=\\p{Lower})(?=\\p{Upper})",
+          "replacement": " "
+        }
+      }
+    }
+  },
+  "mappings": {
+    "my_type": {
+      "properties": {
+        "text": {
+          "type": "text",
+          "analyzer": "my_analyzer"
+        }
+      }
+    }
+  }
+}
+
+POST my_index/_analyze
+{
+  "analyzer": "my_analyzer",
+  "text": "The fooBarBaz method"
+}
+----------------------------
+// CONSOLE
+
+/////////////////////
+
+[source,js]
+----------------------------
+{
+  "tokens": [
+    {
+      "token": "the",
+      "start_offset": 0,
+      "end_offset": 3,
+      "type": "<ALPHANUM>",
+      "position": 0
+    },
+    {
+      "token": "foo",
+      "start_offset": 4,
+      "end_offset": 6,
+      "type": "<ALPHANUM>",
+      "position": 1
+    },
+    {
+      "token": "bar",
+      "start_offset": 7,
+      "end_offset": 9,
+      "type": "<ALPHANUM>",
+      "position": 2
+    },
+    {
+      "token": "baz",
+      "start_offset": 10,
+      "end_offset": 13,
+      "type": "<ALPHANUM>",
+      "position": 3
+    },
+    {
+      "token": "method",
+      "start_offset": 14,
+      "end_offset": 20,
+      "type": "<ALPHANUM>",
+      "position": 4
+    }
+  ]
+}
+----------------------------
+// TEST[continued]
+// TESTRESPONSE
+
+/////////////////////
+
+The above returns the following terms:
+
+[source,js]
+----------------------------
+[ the, foo, bar, baz, method ]
+----------------------------
+
+Querying for `bar` will find the document correctly, but highlighting on the
+result will produce incorrect highlights, because our character filter changed
+the length of the original text:
+
+[source,js]
+----------------------------
+PUT my_index/my_doc/1?refresh
+{
+  "text": "The fooBarBaz method"
+}
+
+GET my_index/_search
+{
+  "query": {
+    "match": {
+      "text": "bar"
+    }
+  },
+  "highlight": {
+    "fields": {
+      "text": {}
+    }
+  }
+}
+----------------------------
+// TEST[continued]
+// CONSOLE
+
+The output from the above is:
+
+[source,js]
+----------------------------
+{
+  "took": 1,
+  "timed_out": false,
+  "_shards": {
+    "total": 5,
+    "successful": 5,
+    "failed": 0
+  },
+  "hits": {
+    "total": 1,
+    "max_score": 0.4375,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_type": "my_doc",
+        "_id": "1",
+        "_score": 0.4375,
+        "_source": {
+          "text": "The fooBarBaz method"
+        },
+        "highlight": {
+          "text": [
+            "The foo<em>Ba</em>rBaz method" <1>
+          ]
+        }
+      }
+    ]
+  }
 }
---------------------------------------------------
+----------------------------
+<1> Note the incorrect highlight.
+// TESTRESPONSE
```

Comment:
That means it didn't match the // TESTRESPONSE with the snippet above it.
It doesn't like if there is a newline between the snippet and the marker.
Or something like that.
On May 15, 2016 10:59 AM, "Clinton Gormley" <notifications@github.com>
wrote:

> In docs/reference/analysis/charfilters/pattern-replace-charfilter.asciidoc
> <https://github.com/elastic/elasticsearch/pull/18356#discussion_r63292994>
> :
>
> >  }
> > ---------------------------------------------------
> > +----------------------------
> > +<1> Note the incorrect highlight.
> > +// TESTRESPONSE
>
> I get the following exception when running the docs check:
>
> TESTRESPONSE not paired with a snippet at .../charfilters/pattern-replace-charfilter.asciidoc:296
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/18356/files/89bbc7f861ea99c3af0f0db8718be64f7d62db79#r63292994>
>